test/controllers/authorizeService.js

```diff
@@ -0,0 +1,184 @@
+var proxyquire = require('proxyquire').noCallThru();
+
+describe('Accounting Service', function () {
+
+    var DEFAULT_URL = 'http://example/path';
+    var DEFAULT_WSTOREHOST = 'localhost';
+    var DEFAULT_APIKEY = 'apiKey';
+
+    var getAuthorizeServiceController = function (accServiceSchema, config, uuidMock) {
+        return proxyquire('../../controllers/authorizeService', {
+            '../db/schemas/accountingService': accServiceSchema,
+            '../config': config,
+            'node-uuid': uuidMock
+        }).authorizeService;
+    };
+
+    var invalidRequest = function (handler, ip, body, statusExpected, bodyExpected, done) {
+        var config = {
+            appHost: DEFAULT_WSTOREHOST
+        };
+
+        var req = {
+            ip: {
+                replace: function (expr) { return ip}
+            },
+            body: body
+        };
+
+        var accSerivceController = getAuthorizeServiceController({}, config, {});
+
+        var res = jasmine.createSpyObj('res', ['status', 'json']);
+
+        res.status.and.returnValue(res);
+
+        accSerivceController[handler](req, res);
+
+        setTimeout(function () {
+
+            expect(res.status).toHaveBeenCalledWith(statusExpected);
+            expect(res.json).toHaveBeenCalledWith(bodyExpected);
+
+            done();
+
+        }, 100);
+    };
+
+    describe('Get api-key', function () {
+
+        it('should return 401 when the requester is not the WStore', function (done) {
+
+            invalidRequest('getApiKey', '130.25.13.12', undefined, 401, {error: 'Invalid remote client'}, done);
+        });
+
+        it('should return 400 when the body is empty', function (done) {
+
+            invalidRequest('getApiKey', DEFAULT_WSTOREHOST, undefined, 400, {error: 'Invalid body'}, done);
+        });
+
+        it('should return 400 when the "url" is not defined', function (done) {
+
+            invalidRequest('getApiKey', DEFAULT_WSTOREHOST, '{}', 400, {error: 'Url missing'}, done);
+        });
+
+        var saveAccountingService = function (saveReturn, sendFunction, sendMessage, statusExpected, done) {
```

Comment:
The first parameter and the second parameter are related. When `saveReturn` is `null`, `sendFunction` is `json`. Maybe you can use a boolean to define if the save function is going to fail or not and then check that `json` or `send` have been called appropriately.